### PR TITLE
k-shape and k-means added. Its corresponding test has been added too

### PR DESCRIPTION
**Pull request by dcuestam**
Make sure you have checked _all_ steps below.

### Description
- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [ ] My commits have been squashed if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

## License
- [ ] Add a [Mozilla Public License 2.0 license header](http://mozilla.org/MPL/2.0/) to all the new files.

### Documentation
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.

### CODE REVIEW

File: R/clustering.R [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+#Copyright (c) 2018 Shapelets.io
+#
+#This Source Code Form is subject to the terms of the Mozilla Public
+#License, v. 2.0. If a copy of the MPL was not distributed with this
+#file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#' KMeans
+#'
+#' Calculates the K-Means algorithm.
+#'
```
Comment (avilchess) [inline R/clustering.R:11]: Set reference to paper

File: R/clustering.R [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      list("centroids" = createArray(out$centroids),
+           "labels" = createArray(out$labels))
+    
+    return(newList)
+  }
+
+#' KShape
+#'
+#' Calculates the k-shape algorithm.
+#'
```
Comment (avilchess) [inline R/clustering.R:51]: Set reference to paper

**Review by avilchess — CHANGES_REQUESTED** [review]

**Review by avilchess — DISMISSED** [review]

File: R/clustering.R [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+#' Calculates the K-Means algorithm.
+#' 
+#' [1] S. Lloyd. 1982. Least squares quantization in PCM. IEEE Transactions on Information Theory, 28, 2,
+#' Pages 129-137.
+#'
+#' @param tss Expects an input array whose dimension zero is the length of the time series (all the same) and
+#' dimension one indicates the number of time series.
+#' @param k The number of means to be computed.
+#' @param tolerance The error tolerance to stop the computation of the centroids.
+#' @param max_iterations The maximum number of iterations allowed.
```
Comment (otorreno) [inline R/clustering.R]: maxIterations

File: R/clustering.R [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+#' Calculates the k-shape algorithm.
+#'
+#' [1] John Paparrizos and Luis Gravano. 2016. k-Shape: Efficient and Accurate Clustering of Time Series.
+#' SIGMOD Rec. 45, 1 (June 2016), 69-76.
+#'
+#' @param tss Expects an input array whose dimension zero is the length of the time series (all the same) and
+#' dimension one indicates the number of time series.
+#' @param k The number of means to be computed.
+#' @param tolerance The error tolerance to stop the computation of the centroids.
+#' @param max_iterations The maximum number of iterations allowed.
```
Comment (otorreno) [inline R/clustering.R]: Same here

**Review by otorreno — CHANGES_REQUESTED** [review]

File: tests/testthat/test_clustering.R [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  expected.centroids <-
+    sort(c(array(as.double(
+      c(
+        -0.5234,
+        0.1560,
+        -0.3627,
+        -1.2764,
+        -0.7781,
+        0.9135,
+        1.8711,-0.7825,
```
Comment (avilchess) [inline tests/testthat/test_clustering.R]: David, please keep 1 element per line.

File: tests/testthat/test_clustering.R [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        1.5990,
+        0.1701,
+        0.4082,
+        0.8845,
+        -1.4969,
+        -0.7825,-0.6278,
```
Comment (avilchess) [inline tests/testthat/test_clustering.R]: idem.

**Review by avilchess — CHANGES_REQUESTED** [review]

**Review by avilchess — DISMISSED** [review]

**Review by otorreno — APPROVED** [review]
LGTM

**Pull request merged**